vevestaX/vevesta.py

```diff
@@ -828,3 +828,34 @@ def commit(self, techniqueUsed, filename=None, message=None, version=None, proje
             print("Wrote experiment to tool, Vevesta")
         else:
             print("Failed to write experiment to tool, Vevesta")
+
+    def git_commit(self, repoName, sourceBranch='main', newBranch=None, commitMessage=None):
+        if newBranch is None:
+            newBranch = sourceBranch
+
+        git_access_token = open('git_access_token.txt', "r").read()
+        g = Github(git_access_token)
+        repo = g.get_repo(repoName)
+        sb = repo.get_branch(sourceBranch)
+
+        ipynb_file_name = self.get_filename()
+        if os.path.exists(ipynb_file_name):
+            ipynb_file_content = open(ipynb_file_name, 'r').read()
+            try:
+                contents = repo.get_contents(ipynb_file_name, ref=sourceBranch)
+                if commitMessage is None:
+                    commitMessage = 'updated ' + ipynb_file_name
```

Comment:
the technique used in the commit function will be used to create the branch name and the message will be commit message